src/client/notes.rs

```diff
@@ -101,80 +105,176 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store, A: TransactionAuthenticator> Client
+
+                let account_tags = maybe_await!(self.get_account_stubs())?
+                    .into_iter()
+                    .map(|(stub, _)| NoteTag::from_account_id(stub.id(), NoteExecutionHint::Local))
+                    .collect::<Result<Vec<_>, _>>()?;
+
+                let uncommited_note_tags =
+                    maybe_await!(self.get_input_notes(NoteFilter::Expected))?
+                        .into_iter()
+                        .filter_map(|note| note.metadata().map(|metadata| metadata.tag()))
+                        .collect::<Vec<_>>();
+
+                let ignored =
+                    ![tracked_tags, account_tags, uncommited_note_tags].concat().contains(&tag);
+
+                if ignored {
+                    info!("Ignoring note with tag {}", tag);
                 }
-            }
 
-            Some(built_inclusion_proof)
-        } else {
-            None
+                let record_details = NoteRecordDetails::new(
+                    details.nullifier().to_string(),
+                    details.script().clone(),
+                    details.inputs().values().to_vec(),
+                    details.serial_num(),
+                );
+
+                InputNoteRecord::new(
+                    details.id(),
+                    details.recipient().digest(),
+                    details.assets().clone(),
+                    NoteStatus::Expected { created_at: 0 },
+                    None,
+                    None,
+                    record_details,
+                    ignored,
+                    Some(tag),
+                )
+            },
+            NoteFile::NoteWithProof(note, inclusion_proof) => {
+                let details = NoteRecordDetails::new(
+                    note.nullifier().to_string(),
+                    note.script().clone(),
+                    note.inputs().values().to_vec(),
+                    note.serial_num(),
+                );
+
+                InputNoteRecord::new(
+                    note.id(),
+                    note.recipient().digest(),
+                    note.assets().clone(),
+                    NoteStatus::Committed {
+                        block_height: inclusion_proof.origin().block_num as u64,
+                    },
+                    Some(*note.metadata()),
+                    Some(inclusion_proof),
+                    details,
+                    false,
+                    None,
+                )
+            },
         };
+        let id = note.id();
 
-        let note = InputNoteRecord::new(
-            note.id(),
-            note.recipient(),
-            note.assets().clone(),
-            note.status(),
-            note.metadata().copied(),
-            inclusion_proof,
-            note.details().clone(),
-        );
-
-        maybe_await!(self.store.insert_input_note(&note)).map_err(|err| err.into())
+        maybe_await!(self.store.insert_input_note(note))
+            .map_err(<StoreError as Into<ClientError>>::into)?;
```

Comment:
is the `map_err` needed?

```suggestion
        maybe_await!(self.store.insert_input_note(note))?;
```